### PR TITLE
feat: add base implementation of linuxX64 target

### PR DESCRIPTION
**Pull request by lauzadis**
This PR adds a basic, empty implementation of a new KMP target, linuxX64. This is required to unblock our upgrade to Dokka 1.9.0. It also serves as a proof-of-concept of adding a new target to the SDK.

## Issue \#

Related to but does not fully complete [aws-sdk-kotlin#229](https://github.com/awslabs/aws-sdk-kotlin/issues/229)

## Description of changes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/auth/credentials/ImdsCredentialsProviderTest.kt
Comment (aajtodd) [inline aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/auth/credentials/ImdsCredentialsProviderTest.kt:1]: If possible we should strive to keep these tests running for all platforms, refactoring if necessary.

File: aws-runtime/aws-config/native/src/aws/sdk/kotlin/runtime/auth/credentials/SdkIOExceptionNative.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.runtime.auth.credentials
+
+// FIXME KMP implementation
+@Suppress("ACTUAL_WITHOUT_EXPECT")
+internal actual class SdkIOException : Exception()
```
Comment (aajtodd) [inline aws-runtime/aws-config/native/src/aws/sdk/kotlin/runtime/auth/credentials/SdkIOExceptionNative.kt:9]: Do we even need this anymore? We have `IOException` in `runtime-core`...

File: tests/benchmarks/service-benchmarks/jvm/src/aws/sdk/kotlin/benchmarks/service/BenchmarkHarness.kt
Comment (aajtodd) [inline tests/benchmarks/service-benchmarks/jvm/src/aws/sdk/kotlin/benchmarks/service/BenchmarkHarness.kt:1]: What was platform specific about these?

File: aws-runtime/aws-config/native/src/aws/sdk/kotlin/runtime/auth/credentials/SdkIOExceptionNative.kt (hunk 1 above)
Comment (lauzadis) [reply aws-runtime/aws-config/native/src/aws/sdk/kotlin/runtime/auth/credentials/SdkIOExceptionNative.kt:9]: Nope, I removed it, thanks

File: tests/benchmarks/service-benchmarks/jvm/src/aws/sdk/kotlin/benchmarks/service/BenchmarkHarness.kt
Comment (lauzadis) [reply tests/benchmarks/service-benchmarks/jvm/src/aws/sdk/kotlin/benchmarks/service/BenchmarkHarness.kt:1]: The benchmark application definition in `build.gradle.kts`. I got many unresolved symbol errors when running `:tests:benchmarks:service-benchmarks:compileKotlinLinuxX64`. We have this segment of code in the build file which seems JVM-specific.

```
tasks.named<JavaExec>("run") {
    classpath += objects.fileCollection().from(
        tasks.named("compileKotlinJvm"),
        configurations.named("jvmRuntimeClasspath"),
    )
}
```

File: aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/auth/credentials/ImdsCredentialsProviderTest.kt
Comment (lauzadis) [reply aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/auth/credentials/ImdsCredentialsProviderTest.kt:1]: I'm able to keep this in `common` now that I removed `SdkIOException`

File: aws-runtime/aws-config/build.gradle.kts [hunk 2]
```diff
@@ -192,6 +192,9 @@ NOTE: We need the following tasks to depend on codegen for gradle caching/up-to-
 * `compileKotlinJvm` (Type=KotlinCompile)
 * `compileKotlinMetadata` (Type=KotlinCompileCommon)
 * `sourcesJar` and `jvmSourcesJar` (Type=org.gradle.jvm.tasks.Jar)
+*
+* For Kotlin/Native, an additional dependency is introduced:
+* `compileKotlin<Platform>` (Type=KotlinNativeCompile) (e.g. compileKotlinLinuxX64)
```
Comment (aajtodd) [inline aws-runtime/aws-config/build.gradle.kts:197]: is there not an abstract base class of some sort shared by all compilation targets? 🤔

File: aws-runtime/aws-config/common/src/aws/sdk/kotlin/runtime/auth/credentials/ImdsCredentialsProvider.kt [hunk 3]
```diff
@@ -33,8 +34,6 @@ private const val CREDENTIALS_BASE_PATH: String = "/latest/meta-data/iam/securit
 private const val CODE_ASSUME_ROLE_UNAUTHORIZED_ACCESS: String = "AssumeRoleUnauthorizedAccess"
 private const val PROVIDER_NAME = "IMDSv2"
 
-internal expect class SdkIOException : Exception // FIXME move this to the proper place when we do the larger KMP Exception refactor
```
Comment (aajtodd) [inline aws-runtime/aws-config/common/src/aws/sdk/kotlin/runtime/auth/credentials/ImdsCredentialsProvider.kt:36]: nice

**Review by aajtodd — APPROVED** [review]
Looks good barring any changes to templates/accessibility that come about.

File: tests/benchmarks/service-benchmarks/build.gradle.kts [hunk 4]
```diff
@@ -40,7 +40,7 @@ if (missingServices.isEmpty()) {
                 optinAnnotations.forEach { languageSettings.optIn(it) }
             }
 
-            commonMain {
+            jvmMain {
```
Comment (ianbotsf) [inline tests/benchmarks/service-benchmarks/build.gradle.kts:43]: **Question**: Why did service benchmarks need to become JVM only?

**Review by ianbotsf — APPROVED** [review]

File: tests/benchmarks/service-benchmarks/build.gradle.kts (hunk 4 above)
Comment (lauzadis) [reply tests/benchmarks/service-benchmarks/build.gradle.kts:43]: I had many unresolved symbols come up and could not find an easy way to fix them besides moving this to JVM only

https://github.com/awslabs/aws-sdk-kotlin/pull/1058#discussion_r1361050754

File: aws-runtime/aws-config/build.gradle.kts (hunk 2 above)
Comment (lauzadis) [reply aws-runtime/aws-config/build.gradle.kts:197]: I don't think so, [it seems separate](https://github.com/JetBrains/kotlin/blob/50e75c6336b225895e9f2ae6350d6a135bbce1c1/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/native/tasks/KotlinNativeTasks.kt#L305)

File: services/polly/common/test/aws/sdk/kotlin/services/polly/PollyTest.kt [hunk 5]
```diff
@@ -30,23 +30,27 @@ class PollyPresignerTest {
...
 
-        val presignedRequest = PollyClient {
+        val pollyClient = PollyClient {
             region = "us-east-2"
             credentialsProvider = StaticCredentialsProvider {
                 accessKeyId = "AKID"
                 secretAccessKey = "secret"
             }
             httpClient = NoHttpEngine
-        }.use { polly ->
```
Comment (aajtodd) [inline services/polly/common/test/aws/sdk/kotlin/services/polly/PollyTest.kt:40]: You just need to import our [use](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/io/Closeable.kt#L28) extension

**Pull request merged**